elasticsearch/kontena.yml

```diff
@@ -0,0 +1,96 @@
+stack: kontena/elasticsearch
+version: 0.1.0
+description: Elasticsearch
+expose: elastic
+variables:
+  instances:
+    type: integer
+    from:
+      prompt: How many instances?
+  reset_passwords:
+    type: boolean
+    from:
+      prompt: Reset all default passwords during initial stack installation?
+  elastic_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: ELASTIC_PASSWORD
+      random_string: 12
+    to:
+      vault: ELASTIC_PASSWORD
+  kibana_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: KIBANA_PASSWORD
+      random_string: 12
+    to:
+      vault: KIBANA_PASSWORD
+  logstash_password:
+    only_if: reset_passwords
+    type: string
+    from:
+      vault: LOGSTASH_PASSWORD
+      random_string: 12
+    to:
+      vault: LOGSTASH_PASSWORD
+
+  mem_limit:
+    type: integer
+    default: 2048
+    from:
+      prompt: Memory limit (in megabytes).
+
+  jvm_limit:
+    type: integer
+    from:
+      evaluate: ${mem_limit} * 0.8
+
+services:
+  elastic:
+    image: docker.elastic.co/elasticsearch/elasticsearch:5.5.2
+    instances: {{ instances }}
+    stateful: true
+    mem_limit: "{{ mem_limit }}m"
+    environment:
+      cluster.name: ${STACK}
+      network.host: _ethwe_
+      discovery.zen.ping.unicast.hosts: elastic
+      ES_JAVA_OPTS: "-Xms{{ jvm_limit }}m -Xmx{{ jvm_limit }}m"
+    volumes:
+      - elasticdata:/usr/share/elasticsearch/data
+    hooks:
+      post_start:
+        # {% if reset_passwords %}
+        - cmd: while ! (echo > /dev/tcp/$$HOSTNAME/9200) >/dev/null 2>&1; do sleep 1; done
+          name: Wait for elastic to startup
+          instances: {{ instances }}
+          oneshot: true
+        - cmd: curl -u elastic:changeme -XPOST -d '{"password":"'"$$ELASTIC_PASSWORD"'"}' $$HOSTNAME:9200/_xpack/security/user/elastic/_password
+          name: Change default user password
+          instances: {{ instances }}
+          oneshot: true
+        - cmd: curl -u elastic:$$ELASTIC_PASSWORD -XPOST -d '{"password":"'"$$KIBANA_PASSWORD"'"}' $$HOSTNAME:9200/_xpack/security/user/kibana/_password
```

Comment:
Just wondering could we move kibana & logstash hooks to those stacks?

Reply:
IMO no. Elastic stack is supposed to be fully generic Elastic stack, it has really nothing to do with F and K part. If only Fluentd or Kibana would config the passwords, installing only Elastic would leave those passwords to their un-secure defaults.